apps/hash-graph/bin/cli/src/subcommand/snapshot.rs

```diff
@@ -0,0 +1,113 @@
+use std::io::Write;
+
+use clap::Parser;
+use error_stack::{IntoReport, Result, ResultExt};
+use graph::{
+    knowledge::Entity,
+    logging::{init_logger, LoggingArgs},
+    store::{
+        crud::Read, query::Filter, test_graph, test_graph::BlockProtocolModuleVersions,
+        DatabaseConnectionInfo, PostgresStorePool, StorePool,
+    },
+};
+use tokio_postgres::NoTls;
+use type_system::{DataType, EntityType, PropertyType};
+
+use crate::error::GraphError;
+
+#[derive(Debug, Parser)]
+pub struct SnapshotDumpArgs;
+
+#[derive(Debug, Parser)]
+pub enum SnapshotCommand {
+    Dump(SnapshotDumpArgs),
+}
+
+#[derive(Debug, Parser)]
+#[clap(version, author, about, long_about = None)]
+pub struct SnapshotArgs {
+    #[command(subcommand)]
+    pub command: SnapshotCommand,
+
+    #[clap(flatten)]
+    pub log_config: LoggingArgs,
+
+    #[clap(flatten)]
+    pub db_info: DatabaseConnectionInfo,
+}
+
+pub async fn snapshot(args: SnapshotArgs) -> Result<(), GraphError> {
+    let _log_guard = init_logger(&args.log_config);
+
+    let pool = PostgresStorePool::new(&args.db_info, NoTls)
+        .await
+        .change_context(GraphError)
+        .map_err(|err| {
+            tracing::error!("{err:?}");
+            err
+        })?;
+
+    let store = pool
+        .acquire()
+        .await
+        .change_context(GraphError)
+        .map_err(|err| {
+            tracing::error!("{err:?}");
+            err
+        })?;
+
+    match args.command {
+        SnapshotCommand::Dump(_) => {
+            let data_types = Read::<test_graph::OntologyTypeRecord<DataType>>::read(
```

Comment:
Don't have to do as a drive-by right now, but I think we should look at that module structure soon as we discussed. `test_graph` doesn't seem like a great module name, maybe we just call it `snapshot` to be consistent with this or go with `raw_db` or something like that.